app/bundles/LeadBundle/Controller/LeadController.php

```diff
@@ -764,6 +764,171 @@ private function uploadAvatar(Lead $lead)
+        $mainLead  = $model->getEntity($objectId);
+
+        //set the page we came from
+        $page = $this->factory->getSession()->get('mautic.lead.page', 1);
+
+        //set the return URL
+        $returnUrl = $this->generateUrl('mautic_lead_index', array('page' => $page));
+
+        $postActionVars = array(
+            'returnUrl'       => $returnUrl,
+            'viewParameters'  => array('page' => $page),
+            'contentTemplate' => 'MauticLeadBundle:Lead:index',
+            'passthroughVars' => array(
+                'activeLink'    => '#mautic_lead_index',
+                'mauticContent' => 'lead'
+            )
+        );
+
+        if ($mainLead === null) {
+            return $this->postActionRedirect(
+                array_merge(
+                	$postActionVars,
+                    array(
+                        'flashes' => array(
+                            array(
+                                'type'    => 'error',
+                                'msg'     => 'mautic.lead.lead.error.notfound',
+                                'msgVars' => array('%id%' => $objectId)
+                            )
+                        )
+                    )
+                )
+            );
+        } elseif (!$this->factory->getSecurity()->hasEntityAccess(
+            'lead:leads:editown',
+            'lead:leads:editother',
+            $mainLead->getOwner())
+        ) {
+            return $this->accessDenied();
+        } elseif ($model->isLocked($mainLead)) {
+            //deny access if the entity is locked
+            return $this->isLocked($postActionVars, $mainLead, 'lead.lead');
+        }
+
+ 		//Order the leads
+        $orderBy    = $this->factory->getSession()->get('mautic.lead.orderby', 'l.last_active');
+        $orderByDir = $this->factory->getSession()->get('mautic.lead.orderbydir', 'DESC');
+        
+
+        $action = $this->generateUrl('mautic_lead_action', array('objectAction' => 'merge', 'objectId' => $objectId));
+        $form = $this->get('form.factory')->create(
+                    'lead_merge',
+                    array(),
+                    array('action'        => $action)
+                );
+
+        if ($this->request->getMethod() == 'POST') {
+            if (!$this->isFormCancelled($form)) {
+                if ($this->isFormValid($form)) {
+                    $data = $this->request->request->get('lead');
+
+                    //pull the data from the form in order to apply the form's formatting
+                    foreach ($form as $f) {
+                        $name = $f->getName();
+                        if (strpos($name, 'field_') === 0) {
+                            $data[$name] = $f->getData();
+                        }
+                        
+
+                        $secLeadId = $data['lead_to_merge'];
+                        $secLead =  $model->getRepository('MauticLeadBundle:Lead')->getLeadsByUniqueFields(array(),$secLeadId)[0];
```

Comment:
I don't believe this is supported in 5.3 (our minimum).  You'll probably have to convert this to assign to a variable first, then use the bracket notation.